app/models/piece.rb

```diff
@@ -77,7 +77,28 @@ def piece_present?(array)
   end
 
   def invalid_input?(destination_row, destination_col)
-    destination_row > 7 || destination_col > 7
+    destination_row > 7 || destination_col > 7 || destination_row < 0 || destination_col < 0
+  end
+
+  def destination_with_piece_of_same_color?(destination_row, destination_col)
```

Comment:
Only suggestion is changing the name of the method to   spot_taken_and_same_color?  to be more understandable. Up to you and the rest of the team.